compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5016,56 +5014,15 @@ aloadHelper(TR::Node * node, TR::CodeGenerator * cg, TR::MemoryReference * tempM
 
    TR::SymbolReference * symRef = node->getSymbolReference();
    TR::Symbol * symbol = symRef->getSymbol();
-   TR::Node *constNode=(TR::Node *) (node->getSymbolReference()->getOffset());
-
-   // Dynamic literal pool can change loadaddr of static sym ref to iaload
-   // we have to detect this case and allocate collectable/non-collectable
-   // register based on loadaddr logic
-   // the iaload that came from loadaddr will have the following structure
-   //
-   //          iaload <static sym ref>
-   //             aload (literal pool base)
-
+   TR::Node *constNode = reinterpret_cast<TR::Node *>(node->getSymbolReference()->getOffset());
    bool dynLitPoolLoad = false;
-   if ((node->getOpCodeValue() == TR::aloadi) &&
-         symbol->isStatic() &&
-         node->getSymbolReference()->isFromLiteralPool() &&
-         (node->getNumChildren() > 0 && node->getFirstChild()->getOpCode().hasSymbolReference() &&
-         (node->getFirstChild()->getSymbolReference()->isLiteralPoolAddress() ||
-          (!node->getFirstChild()->getSymbol()->isStatic() &&
-           node->getFirstChild()->getSymbolReference()->isFromLiteralPool()))))
-      {
-      dynLitPoolLoad = true;
-
-      if (symbol->isLocalObject())
-         {
-         tempReg= cg->allocateCollectedReferenceRegister();
-         }
-      else
-         {
-         tempReg= cg->allocateRegister();
-         }
-      }
-   //regular-born aload/iaload
-   else
-      {
-      tempReg = cg->allocateRegister();
-
-      // TODO: add "are we a garbage-collected language?" check
-      if (!symbol->isInternalPointer()  &&
-          !symbol->isNotCollected()     &&
-          !symbol->isAddressOfClassObject() &&
-          // LowerTrees transforms unloadable aconsts to iaload <gen. int shadow> / aconst NULL.  All aconsts are never collectable.
-          !(symRef->isLiteralPoolAddress() && (node->getOpCodeValue() == TR::aloadi) && constNode->isClassUnloadingConst()))
-         {
-         tempReg->setContainsCollectedReference();
-         }
-      }
+   tempReg = OMR::Z::TreeEvaluator::checkAndAllocateReferenceRegister(node, cg, dynLitPoolLoad);
```

Comment:
I don't think we should be using `OMR::Z::`, I think this should be `TR::` so we make use to the extensible class mechanism.

Reply:
Fixed.